fonts/generate.py

```diff
@@ -0,0 +1,657 @@
+import base64
+import json
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import xml.etree.ElementTree as Et
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+
+# Define svg namespace
+from typing import Optional
+import logging
+
+from svgpathtools import Path as SvgPath  # type: ignore
+
+SVG_NS = {"svg": "http://www.w3.org/2000/svg"}
+
+SMUFL_HEADER = """/////////////////////////////////////////////////////////////////////////////
+// Name:        smufl.h
+// Author:      Laurent Pugin
+// Created:     2014-2022
+// Copyright (c) Authors and others. All rights reserved.
+/////////////////////////////////////////////////////////////////////////////
+
+/////////////////////////////////////////////////////////////////////////////
+// NOTE: this file was generated by the ./fonts/generate smufl script
+// and should not be edited because changes will be lost.
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef __VRV_SMUFL_H__
+#define __VRV_SMUFL_H__
+
+//----------------------------------------------------------------------------
+
+namespace vrv {{
+
+//----------------------------------------------------------------------------
+// SMUFL glyphs available by default in Verovio
+//----------------------------------------------------------------------------
+
+enum {{
+{smufl_glyph_list}
+}};
+
+/** The number of glyphs for verification **/
+#define SMUFL_COUNT {len_smufl_codes}
+
+}} // namespace vrv
+
+#endif
+"""
+
+FONTFORGE_SCRIPT = (
+    """import os; fontforge.open("{input_fontpath}").generate("{output_fontpath}")"""
+)
+
+FONTFACE_WRAPPER = """@font-face {{
+    font-family: '{fontname}';
+    src: url(data:application/font-woff2;charset=utf-8;base64,{b64encoding}) format('woff2');
+    font-weight: normal;
+    font-style: normal;
+}}"""
+
+
+B64_FONT_LICENSE = """
+This font is licensed under the SIL Open Font License, http://scripts.sil.org/OFL, with 
+Reserved Font Name "{fontname}".
+
+See https://github.com/rism-digital/verovio/blob/develop/fonts/README.md for more information.
```

Comment:
What does the "blob" part in this URL?

Reply:
That's what GitHub URLs look like if you link to a non-branch-specific (e.g., whatever the default branch is) file.